File: AP2.cpp
#include <stdio.h>
using namespace std;

int main()
{ int t;
    scanf("%d",&t);
    while(t--){
    long long tt,tlt,sumT,n,d,a,i;
    scanf("%lld%lld%lld",&tt,&tlt,&sumT);
    n=2*sumT/(tt+tlt);
    printf("%lld\n",n);
    d=(tlt-tt)/(n-5);
    a=tt-2*d;
    for(i=0;i<n;i++){
            printf("%lld",(a+i*d));
            printf(" ");
        }

    }
    return 0;
}

File: AP2.py
"""Not possible using Python time limit too short for it."""
t=input()
for i in xrange(t):
    tt,tlt,sumT=map(int,raw_input().split())
    n=2*sumT/(tt+tlt)
    print n
    d=(tlt-tt)/(n-5)
    a=tt-2*d
    j=0
    for j in xrange(n-1):
        print a,
        a+=d
    print a
    